mozjs/benches/latin1_string_conversion.rs

```diff
@@ -0,0 +1,244 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::ffi::c_void;
+use std::ptr;
+
+use mozjs::conversions::jsstr_to_string;
+use mozjs::glue::{CreateJSExternalStringCallbacks, JSExternalStringCallbacksTraps};
+use mozjs::jsapi::{
+    JSAutoRealm, JS_NewExternalStringLatin1, JS_NewGlobalObject, OnNewGlobalHookOption,
+};
+use mozjs::rooted;
+use mozjs::rust::{JSEngine, RealmOptions, Runtime, SIMPLE_GLOBAL_CLASS};
+
+fn external_string(c: &mut Criterion) {
+    unsafe {
+        let engine = JSEngine::init().unwrap();
+        let runtime = Runtime::new(engine.handle());
+        let context = runtime.cx();
+        let h_option = OnNewGlobalHookOption::FireOnNewGlobalHook;
+        let c_option = RealmOptions::default();
+        rooted!(in(context) let global = JS_NewGlobalObject(
+            context,
+            &SIMPLE_GLOBAL_CLASS,
+            ptr::null_mut(),
+            h_option,
+            &*c_option,
+        ));
+        let _ac = JSAutoRealm::new(context, global.get());
+
+        let latin1_base = "test latin-1 test
+    latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1
+    test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test
+    latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1 test latin-1
```

Comment:
Can you please build this programmatically? It will be easier to read and also you can easily adjust the size if necessary.